core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -709,6 +709,8 @@ class KafkaApis(val requestChannel: RequestChannel,
         partitions.put(tp, new FetchResponse.PartitionData(data.error, data.highWatermark, lastStableOffset,
           data.logStartOffset, data.preferredReadReplica.map(int2Integer).asJava,
           abortedTransactions, data.records))
+        // record the bytes out metrics only when the response is being sent
+        brokerTopicStats.updateBytesOut(tp.topic, fetchRequest.isFromFollower, data.isReassignmentFetch, data.records.sizeInBytes)
```

Comment:
Worth calling out that this metric will no longer record the down-converted message sizes. I take it this isn't a big deal because we don't expect them to differ by much? 

I also wanted to ask whether we would ever need to compress the message into a different format when downconverting (e.g if a 1.0 client which doesn't support compression codec X is trying to read a message by a 2.5 client with compression codec X). In that case, I imagine, the difference may be worthwhile recording